src/Materials/Textures/Packer/packer.ts

```diff
@@ -0,0 +1,761 @@
+            this.options.colnum = this.options.colnum || 8;
+        }
+
+        this.options.updateInputMeshes = this.options.updateInputMeshes || true;
+        this.options.disposeSources = this.options.disposeSources || true;
+        this._disposeList = [];
+
+        this.options.fillBlanks = this.options.fillBlanks || true;
+
+        if (this.options.fillBlanks === true) {
+            this.options.customFillColor = this.options.customFillColor || 'black';
+        }
+
+        this.options.frameSize = this.options.frameSize || 256;
+        this.options.paddingRatio = this.options.paddingRatio || 0.0115;
+
+        this._paddingValue = Math.ceil(this.options.frameSize * this.options.paddingRatio);
+
+        //Make it an even padding Number.
+        if (this._paddingValue % 2 !== 0) {
+            this._paddingValue++;
+        }
+
+        this.options.paddingMode = this.options.paddingMode || TexturePacker.SUBUV_WRAP;
+
+        if (this.options.paddingMode === TexturePacker.SUBUV_COLOR) {
+            this.options.paddingColor = this.options.paddingColor || new Color4(0, 0, 0, 1.0);
+        }
+
+        this.sets = {};
+        this.frames = [];
+
+        /**
+        * Create the promise and then run through the materials on the meshes.
+        */
+        this.promise = new Promise ((resolve, reject) => {
+            try {
+                if (this.meshes.length === 0) {
+                    //Must be a JSON load!
+                    resolve();
+                }
+                let done = 0;
+                const doneCheck = (mat: Material) => {
+                    done++;
+                    //Check Status of all Textures on all meshes, till they are ready.
+                    if (this.options.map) {
+                        for (let j = 0; j < this.options.map.length; j++) {
+                            let index: string = this.options.map[j];
+                            let t: (Texture | DynamicTexture) = (mat as any)[index];
+
+                            if (t !== null) {
+                                if (!(this.sets as any)[this.options.map[j]]) {
+                                    (this.sets as any)[this.options.map[j]] = true;
+                                }
+
+                                if (this.options.disposeSources) {
+                                    this._disposeList.push(t);
+                                }
+                            }
+                        }
+
+                        if (done === this.meshes.length) {
+                            this._createFrames(resolve);
+                        }
+                    }
+                };
+
+                for (let i = 0; i < this.meshes.length; i++) {
+
+                    let mesh = this.meshes[i];
+                    let material: Nullable< Material > = mesh.material;
+
+                    if (!material) {
+                        return new Error('Mesh has no Material assigned!');

```

Comment:
maybe we can just ignore it instead

Reply:
I think this is resolved, let me test it really quick.

Reply:
getting this resolved now, then will do lint fixes